src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionDiagnosticAnalyzer.cs

```diff
@@ -106,6 +106,14 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext, INamedTyp
                 return;
             }
 
+            // If there are compiler diagnostics on the declaration we can't reliably
+            // tell that the refactoring will be accurate, so don't provide any
+            // code diagnostics
+            if (localDeclaration.GetDiagnostics().Any())
```

Comment:
you want to make sure it's only errors.  this will trigger for warnings as well, and we do not want that to limit this feature.

Reply:
I don't think you can suppress compiler errors., so a check probaably isn't necessary.

You should be able to find/add an analyzer that reports warnings and add it to the compilation of the test. See https://github.com/dotnet/roslyn/tree/master/src/Test/Utilities/Portable/Diagnostics/CommonDiagnosticAnalyzers.cs

Reply:
Since I'm only checking the declaration, none of those are valid... Adding a `#warning` above it works as a compiler diagnostic. TIL

Reply:
> Since I'm only checking the declaration, none of those are valid... 

the primary issue i have is that you're just disabling due to a syntactic diagnostic.  But a syntactic diagnostic doesn't mean there is a problem. it just means a diagnostic placed on the syntax.  if you want to say "disbale when there's something wrong syntactically" then you need to check if it's actually an error.